lib/index.js

```diff
@@ -151,13 +151,107 @@ Appboy.prototype.initializeV2 = function(customEndpoint) {
   window.appboy.initialize(options.apiKey, config);
   
   if (options.automaticallyDisplayMessages) window.appboy.display.automaticallyShowNewInAppMessages();
-  if (userId) window.appboy.changeUser(userId);
-  
-  window.appboy.openSession();
+  this.load('v2', function() {
+    console.log("IBM_TEST: ready and changing user", userId);
+    var userId = self.analytics.user().id();
+
+    if (userId) {
+      window.appboy.changeUser(userId, self._createMessagingReadyCallback(function() {
+        window.appboy.openSession();
+      }))
+    } else {
+      window.appboy.openSession(self._createMessagingReadyCallback());
+    }
 
-  this.load('v2', this.ready);
+    self.ready();
+  });
 };
 
+Appboy.prototype._createMessagingReadyCallback = function(otherCallback) {
```

Comment:
I'm not sure I completely understand the `otherCallback` parameter here and its necessity.